nemo/collections/nlp/models/text_classification/text_classification_model.py

```diff
@@ -192,10 +242,71 @@ def _setup_dataloader_from_config(self, cfg: DictConfig):
             dataset=dataset,
             batch_size=cfg.batch_size,
             shuffle=cfg.shuffle,
-            num_workers=self.dataset_cfg.get("num_workers", 2),
-            pin_memory=self.dataset_cfg.get("pin_memory", False),
-            drop_last=self.dataset_cfg.get("drop_last", False),
-            collate_fn=dataset.collate_fn,  # it is necessary for type checking to be working even if collate_fn is not used
+            num_workers=self.dataset_cfg.num_workers,
+            pin_memory=self.dataset_cfg.pin_memory,
+            drop_last=self.dataset_cfg.drop_last,
+            collate_fn=dataset.collate_fn,
+        )
+
+    def infer(self, queries: List[str], batch_size: int = None, device=None) -> List[int]:
+        """
+        Get prediction for the queries
+        Args:
+            queries: text sequences
+            batch_size: batch size to use during inference.
+            device: device to perform the inference on, eg. cuda or cpu
+        Returns:
+            all_preds: model predictions
+        """
+        if device is None:
+            device = self.device
+        # store predictions for all queries in a single list
+        all_preds = []
+        mode = self.training
+        try:
+            # Switch model to evaluation mode
+            self.eval()
+            self.to(device)
+            infer_datalayer = self._setup_infer_dataloader(queries, batch_size)
```

Comment:
The logic of infer is clearly different from the logic of forward/training_step/*_step as it creates DS/DL inside of it, whereas PTL Model decoupled one from the other. Let's discuss that solution

Reply:
Inference is different from training and evaluation and PT does not have such a thing. For example, infer does not need the infer_step, infer_epoch_end_, etc. It is a new thing.

Reply:
Agree. But I think we should try to standardize when adding new features...

Reply:
This looks similar to ASRModel's ".transcribe()", punctuation model's "add_punctuation_capitalization()" ... 
Such functions are really useful for notebooks and prototyping.

But I agree that "infer" name is a kind of generic and could lead to confusion, @VahidooX  how about renaming it to something model-specific, like "classify_text"?

Reply:
Looks a good idea! Would update it to classify_text().